### ISSUE TITLE
Mac OS 12 Compile Issues

### ISSUE THREAD

**Issue by No0n3Left** [goatpig/BitcoinArmory#330]
Here are the errors:
```
--------------------------------------------------------------------------------
ERROR: coercing to Unicode: need string or buffer, list found
--------------------------------------------------------------------------------

********************************************************************************
2017-Oct-16 09:01pm
Executing: "make -j4 2>&1 | tee -a /Users/name/Downloads/armory_0.96.3-src/osxbuild/build-app.log.txt"
Executing from: "[]"

--------------------------------------------------------------------------------
ERROR: coercing to Unicode: need string or buffer, list found
--------------------------------------------------------------------------------

Traceback (most recent call last):
  File "build-app.py", line 741, in <module>
    main()
  File "build-app.py", line 135, in main
    compile_libpng()
  File "build-app.py", line 466, in compile_libpng
    src = path.join(pngDir, '.libs', dylib)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/posixpath.py", line 70, in join
    elif path == '' or path.endswith('/'):
AttributeError: 'list' object has no attribute 'endswith'
```

**Comment by droark** [comment]
Do you have Unicode characters in your username (e.g., Ni**ñ**a)? Python tends to have issues with it, to the point that Armory won't accept file paths with Unicode chracters.

**Comment by zalmarge** [comment]
I've got to the same error today while building Armory on macOS 10.13. The problem is actually with a libpng URL. 

The build script build-app.py downloads https://sourceforge.net/projects/libpng/files/libpng16/1.6.31/libpng-1.6.31.tar.xz – which is just an HTML error page stating, that "file is not found". Unpack function silently fails unpacking it and pngDir is passed as an empty list to path.join().

Fixed it by changing `libpngVer     = '1.6.31'`to `libpngVer     = '1.6.34'` in build-app.py.
